### PR TITLE
Preparation for sending ARKit tracking state statistics

### PR DESCRIPTION
**Pull request by yklishevich**
PR for https://linear.app/fantasmo/issue/MW-88/monitor-and-accumulate-ar-problems

### CODE REVIEW

File: FantasmoSDK/Classes/Network/FMApi.swift [hunk 1]
```diff
@@ -168,6 +168,16 @@ class FMApi {
             error: postError)
     }
     
+    
+    func send(arKitTrackingStateStatistics: ARKitTrackingStateStatistics) {
```
Comment (lucaskuzma) [inline FantasmoSDK/Classes/Network/FMApi.swift:172]: We do not know the API situation yet. This should not be part of this PR.

File: FantasmoSDK/Classes/ValueTypes/ARKitTrackingStateStatistics.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    private(set) var framesWithNormalTrackingState: Int = 0
+    
+    private(set) var totalNumberOfFrames: Int = 0
+    
+    mutating func update(with trackingState: ARCamera.TrackingState) {
+        totalNumberOfFrames += 1
+        
+        switch trackingState {
+        case .limited(let reason):
+            framesWithLimitedTrackingStateByReason[reason]! += 1
```
Comment (lucaskuzma) [inline FantasmoSDK/Classes/ValueTypes/ARKitTrackingStateStatistics.swift:34]: Can we get rid of the forced unwrap?

File: FantasmoSDK/Classes/ValueTypes/ARKitTrackingStateStatistics.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
+//
+//  FrameStatistics.swift
+//  FantasmoSDK
+//
+//  Created by Yauheni Klishevich on 13.07.2021.
+//
+
+import ARKit
+
+/// Statistics on frames by the quality of ARKit posion tracking.
```
Comment (lucaskuzma) [inline FantasmoSDK/Classes/ValueTypes/ARKitTrackingStateStatistics.swift:10]: typo in `position`

**Review by lucaskuzma — CHANGES_REQUESTED** [review]

File: FantasmoSDK/Classes/Network/FMApi.swift (hunk 1 above)
Comment (yklishevich) [reply FantasmoSDK/Classes/Network/FMApi.swift:172]: Ok, I have removed the method

File: FantasmoSDK/Classes/ValueTypes/ARKitTrackingStateStatistics.swift (hunk 2 above)
Comment (yklishevich) [reply FantasmoSDK/Classes/ValueTypes/ARKitTrackingStateStatistics.swift:34]: I guess you propose to write:

`framesWithLimitedTrackingStateByReason[reason] = (framesWithLimitedTrackingStateByReason[reason] ?? 0) + 1`

and this will work stably in any circumstances (whether `framesWithLimitedTrackingStateByReason` is prepopulated with 0s or not).

Ok, agree

File: FantasmoSDK/Classes/ValueTypes/ARKitTrackingStateStatistics.swift (hunk 3 above)
Comment (yklishevich) [reply FantasmoSDK/Classes/ValueTypes/ARKitTrackingStateStatistics.swift:10]: I really like such comments - thanks!
